### PR TITLE
Update the Angular wrapper to work with Angular@16.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
The main changes made in this PR:
1. Since the release of [angular@16](https://www.npmjs.com/package/@angular/cli/v/16.0.0) the `@handsontable/angular` package was no longer compatible with the latest Angular release. This PR aims to change that.
    I moved the current wrapper logic to a new library template based on `@angular/cli@12`. The new wrapper build uses the Ivy compiler, and being built with `angular@12` makes it compatible with Angular versions 12 up to 16 (currently the latest release).
    
    **Unfortunately, it breaks the compatibility with Angular versions 11 and earlier.** 
    It's worth noting, that [Angular currently provides active support only for versions 14+](https://angular.io/guide/releases#actively-supported-versions).

2. Currently, the examples workflow does not test the angular-based examples properly. 
    When building the `/next/` section of the examples [only `handsontable` is being symlinked in the examples' `node_modules`](https://github.com/handsontable/handsontable/actions/runs/5070569056/jobs/9105914867#step:20:27). What's even worse, [the wrapper version being used in the build and testing process is not `latest`, but `12.0.1`.](https://github.com/handsontable/handsontable/blob/b7cca4520a7284727d9b878a9e01cb695263b0bc/examples/next/docs/angular/demo/package.json#L29)
    
   After the changes made in this PR, [both `handsontable` and `@handsontable/angular` should be linked properly in the examples build/test workflow.](https://github.com/handsontable/handsontable/actions/runs/5197296330/jobs/9372056945#step:20:18)

3. I removed the Angular 9-11 examples, as these versions are no longer compatible with the wrapper.
4. I added Angular 14-16 examples + created an `angular-next` section which builds and tests the wrapper against the version of Angular tagged as `next`.
5. While rewriting the wrapper to later versions of Angular (mostly 16 and 12), I had to make some corrections regarding the typescript typing. There are places in the wrapper logic that would benefit from a bigger refactor, but I did not make any major changes as it was not in the scope of this task.
6. For the same reason as above, I had to make some changes to Handsontable, namely, adding some missing type definitions that were not tested with the wrapper based on Angular@9.
7. I reverted the temporary changes done in the release process of `12.4.0`, which purpose was to prevent the workflows from using the latest `angular` version.
8. I modified the `code-examples` workflow to utilize the `actions/setup-node`'s caching mechanism instead of `actions/cache`. The latter would break the new angular wrapper build + it makes sense to use the pre-built caching that is already being used in our other workflows.
9. I had to make some modifications to other workflows, as the new set of dependencies being introduced with `@angular/cli@12` broke some things, like the documentation linter. This is a bigger issue regarding dependency management in the monorepo, which will have to be tackled at some point in the future.
10. For the same shared-dependency-based reasons, I had to modify the `vue3` wrapper tests, as some of them no longer passed. (The changes make sense, as the types were mismatched in one of the tests, but it was not being analyzed before)

### How has this been tested?
- Tested the new wrapper manually with Angular versions 12 through 16 + `next`.
- [The `code-examples` workflow built the examples for the same versions on the CI.](https://github.com/handsontable/handsontable/actions/runs/5197296330)

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. handsontable/dev-handsontable#1292
2. #10349 

### Affected project(s):
- [x] `handsontable`
- [x] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: examples/next/docs/angular-next/basic-example/spec/support/jasmine.config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    colors: 1,
+    cleanStack: 1,
+    verbosity: 4,
+    listStyle: 'indent',
+    timeUnit: 'ms',
+    timeThreshold: { ok: 500, warn: 1000, ouch: 3000 },
+    activity: false,
+    emoji: true,
+    beep: true
+});
```
Comment (budnix) [inline examples/next/docs/angular-next/basic-example/spec/support/jasmine.config.js:29]:
```suggestion
jasmine.loadConfig({
  spec_dir: 'spec',
  spec_files: ['**/*[sS]pec.js'],
  helpers: ['helpers/**/*.js'],
  random: true,
  stopSpecOnExpectationFailure: false
});
jasmine.jasmine.DEFAULT_TIMEOUT_INTERVAL = 15000;

// setup terminal reporter
const JasmineReporter = require('jasmine-console-reporter');
const reporter = new JasmineReporter({
  colors: 1,
  cleanStack: 1,
  verbosity: 4,
  listStyle: 'indent',
  timeUnit: 'ms',
  timeThreshold: { ok: 500, warn: 1000, ouch: 3000 },
  activity: false,
  emoji: true,
  beep: true
});
```

File: examples/next/docs/angular-next/basic-example/src/app/app.component.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div id="example">
+  <hot-table [settings]="hotSettings"></hot-table>
+<!--  <lib-hot-table></lib-hot-table>-->
```
Comment (budnix) [inline examples/next/docs/angular-next/basic-example/src/app/app.component.html:3]: 🤔

File: examples/next/docs/angular/basic-example/package.json [hunk 3]
```diff
@@ -18,33 +18,33 @@
...
-    "@angular-devkit/build-angular": "latest",
-    "@angular/localize": "latest",
-    "@angular/service-worker": "latest",
-    "@angular/cli": "latest",
-    "@angular/compiler-cli": "latest",
+    "@angular-devkit/build-angular": "16",
+    "@angular/localize": "16",
+    "@angular/service-worker": "16",
+    "@angular/cli": "16",
+    "@angular/compiler-cli": "16",
```
Comment (budnix) [inline examples/next/docs/angular/basic-example/package.json:40]:
```suggestion
    "@angular-devkit/build-angular": "^16.0.0",
    "@angular/localize": "^16.0.0",
    "@angular/service-worker": "^16.0.0",
    "@angular/cli": "^16.0.0",
    "@angular/compiler-cli": "^16.0.0",
```

File: examples/next/docs/angular/basic-example/src/app/app.component.html [hunk 4]
```diff
@@ -1,3 +1,4 @@
 <div id="example">
   <hot-table [settings]="hotSettings"></hot-table>
+<!--  <lib-hot-table></lib-hot-table>-->
```
Comment (budnix) [inline examples/next/docs/angular/basic-example/src/app/app.component.html:3]: 🤔

**Review by budnix — CHANGES_REQUESTED** [review]

File: examples/next/docs/angular-14/basic-example/src/app/app.component.html [hunk 5]
```diff
@@ -1,3 +1,4 @@
 <div id="example">
   <hot-table [settings]="hotSettings"></hot-table>
+<!--  <lib-hot-table></lib-hot-table>-->
```
[the remark this answers is not in the document]
Comment (jansiegel) [reply examples/next/docs/angular-14/basic-example/src/app/app.component.html:3]: Leftover 👍

**Review by budnix — DISMISSED** [review]
Looks good 👌 I tested the bundle [0.0.0-next-44d778b-20230607](https://www.npmjs.com/package/@handsontable/angular/v/0.0.0-next-44d778b-20230607) with simple apps that use Angular 12 and Angular 16, and I didn't find any issues.

File: examples/next/docs/angular-16/demo/package.json [hunk 6]
```diff
@@ -18,33 +18,33 @@
...
-    "@angular/router": "v9-lts",
-    "@handsontable/angular": "12.0.1",
+    "@angular/animations": "16",
+    "@angular/common": "16",
+    "@angular/compiler": "16",
+    "@angular/core": "16",
+    "@angular/forms": "16",
+    "@angular/platform-browser": "16",
+    "@angular/platform-browser-dynamic": "16",
+    "@angular/router": "16",
```
[the remark this answers is not in the document]
Comment (budnix) [reply examples/next/docs/angular-16/demo/package.json:28]: @jansiegel, what about that?

File: examples/next/docs/angular/basic-example/package.json [hunk 7]
```diff
@@ -18,33 +18,33 @@
...
-    "@angular/router": "latest",
-    "@handsontable/angular": "12.0.1",
+    "@angular/animations": "^16.0.0",
+    "@angular/common": "^16.0.0",
+    "@angular/compiler": "^16.0.0",
+    "@angular/core": "^16.0.0",
+    "@angular/forms": "^16.0.0",
+    "@angular/platform-browser": "^16.0.0",
+    "@angular/platform-browser-dynamic": "^16.0.0",
+    "@angular/router": "^16.0.0",
```
Comment (budnix) [inline examples/next/docs/angular/basic-example/package.json:28]: For `/next/` examples, shouldn't the "latest" tag be used here?

Comment (jansiegel) [reply examples/next/docs/angular/basic-example/package.json]: Yup https://github.com/handsontable/handsontable/pull/10396/commits/9cab13b56a5540b0a740bb2d7604c4345bfd5898

File: docs/.vuepress/handsontable-manager/dependencies.js [hunk 8]
```diff
@@ -92,12 +92,13 @@ const buildDependencyGetter = (version) => {
       rxjs: ['https://cdn.jsdelivr.net/npm/rxjs@6/bundles/rxjs.umd.js', [/* todo */]],
       'core-js': ['https://cdn.jsdelivr.net/npm/core-js@2/client/core.min.js', [/* todo */]],
       zone: ['https://cdn.jsdelivr.net/npm/zone.js@0.9/dist/zone.min.js', [/* todo */]],
```
[the remark this answers is not in the document]
Comment (jansiegel) [reply docs/.vuepress/handsontable-manager/dependencies.js]: Seems like a good idea 👍 https://github.com/handsontable/handsontable/pull/10396/commits/80e131154ff3019cb77253b21135f08eb1a08317

**Review by budnix — APPROVED** [review]
Looks good. Also, I've checked the Docs by deploying the latest changes to the staging, and it also looks good.

**Review by aninde — COMMENTED** [review]
I have verified that the build process using `npm run all build` successfully builds the correct version and examples, including Angular 16.0.3 (angular-16 and currently angular-next). 
Screenshot 2023-06-12 at 14 24 25

The Handsontable-Angular integration is functional in environments built with Angular versions 12, 13, 14, 15 and 16. Tested with [0.0.0-next-44d778b-20230607](https://www.npmjs.com/package/@handsontable/angular/v/0.0.0-next-44d778b-20230607).

**Pull request merged**